stateMutators.js

```diff
@@ -0,0 +1,111 @@
+const gameState = {
+    basketPos: 50,
+    basketValue: 0,
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const moveBasketLeftKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos - value,
+});
+
+const moveBasketRightKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos + value,
+});
+
+const moveBasketMouse = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const setBasketValue = (state, value) => ({
+    ...state,
+    basketValue: value,
+});
+
+const resetBasketValue = (state) => ({
+    ...state,
+    basketValue: 0,
+});
+
+const setScore = (state, value) => ({
+    ...state,
+    score: value,
+});
+
+const resetScore = (state) => ({
+    ...state,
+    score: 0,
+});
+
+const setLivesRemaining = (state, value) => ({
+    ...state,
+    livesRemaining: value,
+});
+
+const resetLivesRemaining = (state) => ({
+    ...state,
+    livesRemaining: 3,
+});
+
+const setGameLevel = (state, value) => ({
+    ...state,
+    gameLevel: value,
+});
+
+const resetGameLevel = (state) => ({
+    ...state,
+    gameLevel: 0,
+});
+
+const setGameMode = (state, value) => ({
+    ...state,
+    gameMode: value,
+});
+
+function addFallingObject (state, fallingObj) {
+    state.fallingObjects.push(fallingObj)
+};
+
+function removeFallingObject (state, fallingObj) {
+    let objectIndex = state.fallingObjects.indexOf(fallingObj)
+    state.fallingObjects.splice(objectIndex, 1)
+};
+
+
+// Object constructor for falling object
+function FallingObject(posX, posY, numerator, denominator) {
+    this.posX = posX;
+    this.posY = posY;
+    this.numerator = numerator;
+    this.denominator = denominator;
+    this.value = numerator / denominator
+}
+
+const setFallingObjPosX = (fallingObj, value) => {
```

Comment:
Apart from `posY` I don't think any of these states actually change after creation of the `FallingObject` instance, right? Therefore, I don't think these mutators don't need to exist apart from the one that mutates `posY`.

It would also be good to ensure the first param is `state`, to keep type signature consistency (and return the new state)

Reply:
You're right, I was expecting this to set the position on the x axis, but I forgot it's set in the object constructor. Thanks Jon